server/lomas_server/utils/config.py

```diff
@@ -34,37 +36,55 @@ class Server(BaseModel):
 class DatasetStoreConfig(BaseModel):
     """BaseModel for dataset store configs"""
 
-    ds_store_type: DatasetStoreType
+    ds_store_type: Literal[DatasetStoreType.BASIC.value]  # type: ignore
 
 
 class LRUDatasetStoreConfig(DatasetStoreConfig):
     """BaseModel for dataset store configs in case of a LRU dataset store"""
 
+    ds_store_type: Literal[DatasetStoreType.LRU]  # type: ignore
     max_memory_usage: int
 
 
 class DBConfig(BaseModel):
     """BaseModel for database type config"""
 
-    db_type: str = AdminDBType
+    # db_type: str = AdminDBType
 
 
 class YamlDBConfig(DBConfig):
     """BaseModel for dataset store configs  in case of a Yaml database"""
 
+    db_type: Literal[AdminDBType.YAML]  # type: ignore
     db_file: str
 
 
 class MongoDBConfig(DBConfig):
     """BaseModel for dataset store configs  in case of a  MongoDB database"""
 
+    db_type: Literal[AdminDBType.MONGODB]  # type: ignore
     address: str
     port: int
     username: str
     password: str
     db_name: str
 
 
+class PrivateDBCredentials(BaseModel):
+    """BaseModel for private database credentials."""
+
+
+class S3CredentialsConfig(PrivateDBCredentials):
+    """BaseModel for S3 database credentials."""
+
+    model_config = ConfigDict(extra="allow")
+
+    db_type: Literal[PrivateDatabaseType.S3]  # type: ignore
```

Comment:
Maybe it is just style but is it worth it to add 'Litteral' in front of constants already defined as constants in StrEnums ? no strong opinion here

Reply:
I think we have to do it here to make sure pydantic understands that db_type is a constant Literal with the value PrivateDatabaseType.S3.

As for the type ignore, mypy supports Literals with enums, but not StrEnums. I did not want to change PrivateDatabaseType to be standard enums, so chose to ignore the type error instead.

Reply:
ok 👍🏼